### PR TITLE
(WIP) Implemented splitted ChatList/View

### PR DESCRIPTION
**Pull request by Jikstra**
Just a quick change/hack to get into desktop/react development. Some questions i have:
- where do we want to style the different components? Only in the conversations repo?
- If i have to change something in the conversations repo, i always have to make 2 pull requests and to review them, both have to get pulled which is annoying. What's the reason behind the conversations repo? To have it clearly seperated?
- Currently i have the height/width hardcoded but it should be the screen height/width. How can i access the screen height/width? 

I didn't do much with react, so in case i'm doing something wrong/in a bad way, please please tell me :)

### CODE REVIEW

File: static/main.css [hunk 1]
```diff
@@ -71,3 +73,15 @@ ul {
...
+
+.ChatList {
+  width: 30%;
+  float: left;
+}
+
+.ChatView {
+  width: 70%;
+  float: right;
+  height: 800px;
```
Comment (ralphtheninja) [inline static/main.css]: Maybe `height: 100%'?

Comment (Jikstra) [reply static/main.css]: Height 100% makes it just as big as it needs to be with a scroll bar but not fitting the window height. There's some property i can get from electron, i will give it a try

**Comment by ralphtheninja** [comment]
> where do we want to style the different components? Only in the conversations repo?

We should be able to style in `static/main.css`?

>  If i have to change something in the conversations repo, i always have to make 2 pull requests and to review them, both have to get pulled which is annoying. What's the reason behind the conversations repo? To have it clearly seperated?

Yes, this is painful. Main reason is that `conversations` needs a completely different build step, since it's based on `TypeScript`. I suggest you hack on your fork of it and pick a branch from that in `package.json` for now. Then we don't have to handle both PRs at the same time to get something changed.

**Comment by Jikstra** [comment]
> Yes, this is painful. Main reason is that conversations needs a completely different build step, since it's based on TypeScript.

Don't we want to use TS in deltachat-desktop too? I think it would be better if we would use consistent written code. I don't think it makes a lot of sense to maintain an extra package of components which maybe no one else will ever use besides us. Still we should seperate logic and view components completely.

File: src/renderer/ScreenController.js [hunk 2]
```diff
@@ -59,11 +59,8 @@ class Home extends React.Component {
       case 'CreateGroup':
         Screen = CreateGroup
         break
-      case 'ChatView':
-        Screen = ChatView
-        break
       default:
-        Screen = ChatList
+        Screen = SplittedChatListAndView
```
Comment (ralphtheninja) [inline src/renderer/ScreenController.js]: Maybe just keep the name as `ChatView` and throw everything else away?

**Comment by ralphtheninja** [comment]
> Don't we want to use TS in deltachat-desktop too?

I don't see the reason why. Mainly because I'm old and haven't used TypeScript :smile:

**Comment by Jikstra** [comment]
> > Don't we want to use TS in deltachat-desktop too?
> 
> I don't see the reason why. Mainly because I'm old and haven't used TypeScript smile Otoh, if others want to use it, please go ahead and don't let me stop you.

Me neither. Don't see any benefits of using ts, also i don't really like the extra time which is needed for the the ts -> js transpile time but i never really used it and don't know it's benefits. Maybe @karissa can give us some insights.

File: static/main.css (hunk 1 above)
Comment (ralphtheninja) [reply static/main.css]: > I don't really want to continue "monkey patching/overwriting" css stuff outside of conversations.

Yet another reason to move conversations into desktop :)

File: src/renderer/ScreenController.js (hunk 2 above)
Comment (Jikstra) [reply src/renderer/ScreenController.js]: I don't have any hard opinions on naming, but maybe it makes more sense to use new new names for the components. I will think about it, maybe i have an idea.

**Comment by hpk42** [comment]
closing this in favor of #178

**Pull request closed without merging**